### ISSUE TITLE
Missing function parameters in type do not throw compiler errrors

### ISSUE THREAD

**Issue by chwoerz** [microsoft/TypeScript#59175]
### 🔎 Search Terms

function parameter type, enforce parameter of function

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about  function types

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ssl=14&ssc=51&pln=2&pc=1#code/FAMwrgdgxgLglgewgAgJIQA5hgUwE4AUcm2AXMgM4x7EDmAlMgN7DJvLAC+wwMAnhhzIAYpFiIIAdTgwAFgAUAhnkUBbHLjwVkAXmT9BCEGhKaA3DyhIqyEOVHR4SaXKUr1m7XoKMdAPmZudmCQ0LCwgHoI5AA9AH5kABVZRRhKWQQwABsAE31ZPAQAd2RFFHxCvGQAIxwoRTAKITkhYiw0jELBPH5kOG08HABHMDhBnJ4DIQdxJESBIW82smQIMFVavF8AqhoIWgtgKwgbEAAmezEnCHnBXWQibcCLKOD45AAJfCEkLL58oSDGBgPAoKb5QpFbRlZAVBB4IA

### 💻 Code

```ts

function Inputer(input: string) {
    
}

type FunctionWithParameters = typeof Inputer;

const f: FunctionWithParameters = () => {}
                                // ^? That should throw an error because the input property is required

type FunctionType = (input: number) => string;

const f2: FunctionType = (i) => {};
//    ^? Here only the return type throws an error```


### 🙁 Actual behavior

The compiler does not throw an error for the missing property

### 🙂 Expected behavior

We should get an error like this:
Expected 1 arguments, but got 0.(2554)
(x, y): An argument for 'input' was not provided.

### Additional information about the issue

_No response_

**Comment by jakebailey** [comment]
> * I reviewed the FAQ for entries about  function types

Are you sure?

https://github.com/microsoft/TypeScript/wiki/FAQ#why-are-functions-with-fewer-parameters-assignable-to-functions-that-take-more-parameters
